### ISSUE TITLE
`flutter upgrade` gives wrong error with detached head: "Unable to upgrade Flutter: no origin repository configured"

### ISSUE THREAD

**Issue by iyuhang** [flutter/flutter#56853]
## Steps to Reproduce

1. upgrade from v1.12.13+hotfix.9 to 1.17.0
2. use flutter downgrade to v1.12.13+hotfix.9
3. run on iOS

- I tried to delete the bin/cache directory and do flutter doctor, but the issue is still there
- and if I run flutter upgrade , it shows Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter,
I tried flutter channel stable , not works. please help me 

## Logs

```
Xcode build done.                                           29.4s
[VERBOSE-2:shell.cc(199)] Dart Error: Can't load Kernel binary: Invalid kernel binary format version.
[VERBOSE-2:engine.cc(198)] Could not prepare to run the isolate.
[VERBOSE-2:engine.cc(137)] Engine not prepare and launch isolate.
[VERBOSE-2:shell.cc(437)] Could not launch engine with configuration.
```

```
[✓] Flutter (Channel unknown, v1.12.13+hotfix.9, on Mac OS X 10.14.6 18G87, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.44.2)
[✓] Connected device (2 available)
```

**Comment by iyuhang** [comment]
I also did this :
git clean -xfd
git stash save --keep-index
git stash drop
git pull
flutter doctor
not works, too

**Comment by iyuhang** [comment]
use 
git checkout 1.17.0
flutter doctor
to upgrade to 1.17.0, then can run on iOS, but still cannot use flutter upgrade:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, v1.17.0, on Mac OS X 10.14.6 18G87, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[!] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    ! CocoaPods 1.6.1 out of date (1.8.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade:
        sudo gem install cocoapods
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.44.2)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.
(base) AnsenMBP-4:flutter apple$ flutter channel
Flutter channels:
  master
  dev
  beta
  stable
(base) AnsenMBP-4:flutter apple$ flutter channel stable
Switching to flutter channel 'stable'...
git: Switched to branch 'stable'
git: Your branch is up to date with 'origin/stable'.
Successfully switched to flutter channel 'stable'.
To ensure that you're on the latest build from this channel, run 'flutter upgrade'
(base) AnsenMBP-4:flutter apple$ flutter channel
Flutter channels:
  master
  dev
  beta
* stable
(base) AnsenMBP-4:flutter apple$ flutter upgrade
Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter
```

**Comment by xhp281** [comment]
I have encountered the same problem and look forward to the solution

**Comment by xhp281** [comment]
I solved it with this：git reset --hard 85e5b210b4a2278b7a4104acc0bf1fa4c3196e49

**Comment by iyuhang** [comment]
I checked v1.12.13+hotfix.9 head is f139b11, 
1.17.0 is e6b34c2
I did git reset --hard e6b34c2
and run flutter upgrade

the issue still exists:

`Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter`

**Comment by iyuhang** [comment]
hi @xhp281 
can you run v1.12.13 on iOS now?

**Comment by guopeng1994** [comment]
i saw this:https://flutter.dev/docs/development/ios-project-migration
looks like high version of flutter(>= v1.15.3) changed the framework, and i did see my xcode, there is not app.framework and flutter.framework, after i tried a lot of works by running flutter commands, so flutter downgrade command is not work any more...... 

so i created a new project(v1.12.13+hotfix.9), and find a way to reset my old project (set app.framework and flutter.framework back)

and it worked! here are my steps:

1. find app.framework and flutter.framework in your projectName/ios/Flutter/
    drag them to there:
![image](https://user-images.githubusercontent.com/30316642/81560955-fd412700-93c4-11ea-8fce-5e3bd34dfd22.png)
2. check your [Build Phases], make sure they are in [Link Binary with Libraries] and [Embed frameworkd]
3. still in [Build Phases], check your [Thin Binary],make sure the value is :
/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" thin


after that clean the build dir and run.

yeah, that's it!

the question is : i'm not sure if you should change your [Build Settings> Other Linker Flags] config, and i checked mine, there is no [Flutter] but ['Flutter'], so if it did not work, you may edit it!

hopes it can help u.  :）oops. bad english

**Comment by VladyslavBondarenko** [comment]
Thank you @guopeng1994!
@iyuhang does it help?

**Comment by iyuhang** [comment]
Its real detailed,  much thanks to @guopeng1994 
while I have given up to downgrade now, and try to fix my project issues in new flutter version
Now the main problem is I cannot run flutter upgrade in 1.17.0

run flutter upgrade

shows:

`Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter`

tried:
1. flutter channel stable
2.git clean -xfd
   git stash save --keep-index
   git stash drop
   git pull
   flutter doctor
3. git remote set-url origin https://github.com/flutter/flutter
4. git branch --set-upstream-to=origin/stable stable

none of the above works

anything I can do to fix it?

@VladyslavBondarenko @guopeng1994

**Comment by VladyslavBondarenko** [comment]
@iyuhang 
I have seen `Unable to upgrade Flutter: no origin repository configured` when I checkout to another version (and then I have `channel undefined`).
Simply running `flutter channel stable`, `flutter upgrade` helps in this case. 
Didn't it? Do you mean after all the steps in your comment you can still see `Unable to upgrade Flutter: no origin repository configured`?

**Comment by iyuhang** [comment]
@VladyslavBondarenko 
Yes, I did all the steps on comment, and  flutter doctor says :

` ✓] Flutter (Channel stable, v1.17.0, on Mac OS X 10.14.6 18G87, locale en-US)`  It in Channel stable.

but run flutter upgrade, it still print:
`Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter`

**Comment by jmagman** [comment]
@guopeng1994 Your issue is not related to a `flutter downgrade`.  Please file a new issue.  Thanks!

**Comment by jmagman** [comment]
@iyuhang What is the output of:
```
$ cd /Users/apple/Flutter/flutter
$ git remote show
$ git remote show origin
```

**Comment by iyuhang** [comment]
@jmagman 
```
(base) AnsenMBP-4:flutter apple$ git remote show
origin

(base) AnsenMBP-4:flutter apple$ git remote show origin
* remote origin
  Fetch URL: https://github.com/flutter/flutter
  Push  URL: https://github.com/flutter/flutter
  HEAD branch: master
  Remote branches:
    beta                                                                    tracked
    dev                                                                     tracked
    flutter-1.17-candidate.0                                                tracked
    flutter-1.17-candidate.1                                                tracked
    flutter-1.17-candidate.2                                                tracked
    flutter-1.17-candidate.3                                                tracked
    flutter-1.17-candidate.4                                                tracked
    flutter-1.17-candidate.5                                                tracked
    flutter-1.18-candidate.0                                                tracked
    flutter-1.18-candidate.1                                                tracked
    flutter-1.18-candidate.10                                               tracked
    flutter-1.18-candidate.11                                               tracked
    flutter-1.18-candidate.12                                               tracked
    flutter-1.18-candidate.13                                               tracked
    flutter-1.18-candidate.2                                                tracked
    flutter-1.18-candidate.3                                                tracked
    flutter-1.18-candidate.4                                                tracked
    flutter-1.18-candidate.5                                                tracked
    flutter-1.18-candidate.6                                                tracked
    flutter-1.18-candidate.7                                                tracked
    flutter-1.18-candidate.8                                                tracked
    flutter-1.18-candidate.9                                                tracked
    flutter-1.19-candidate.0                                                tracked
    flutter-1.19-candidate.1                                                tracked
    guidezpl-patch-1                                                        tracked
    master                                                                  tracked
    refs/remotes/origin/CaseyHillers-patch-1                                stale (use 'git remote prune' to remove)
    refs/remotes/origin/DaveShuckerow-patch-1                               stale (use 'git remote prune' to remove)
    refs/remotes/origin/GaryQian-patch-1                                    stale (use 'git remote prune' to remove)
    refs/remotes/origin/GaryQian-patch-1-1                                  stale (use 'git remote prune' to remove)
    refs/remotes/origin/Hixie-patch-1                                       stale (use 'git remote prune' to remove)
    refs/remotes/origin/Hixie-patch-2                                       stale (use 'git remote prune' to remove)
    refs/remotes/origin/Hixie-patch-3                                       stale (use 'git remote prune' to remove)
    refs/remotes/origin/Hixie-patch-4                                       stale (use 'git remote prune' to remove)
    refs/remotes/origin/auth_codes                                          stale (use 'git remote prune' to remove)
    refs/remotes/origin/dart_flags                                          stale (use 'git remote prune' to remove)
    refs/remotes/origin/data_table_checkboxes                               stale (use 'git remote prune' to remove)
    refs/remotes/origin/dependabot/bundler/dev/ci/docker_linux/excon-0.71.0 stale (use 'git remote prune' to remove)
    refs/remotes/origin/dependabot/bundler/dev/ci/mac/excon-0.71.0          stale (use 'git remote prune' to remove)
    refs/remotes/origin/dnfield-patch-1                                     stale (use 'git remote prune' to remove)
    refs/remotes/origin/driver-frame-sync                                   stale (use 'git remote prune' to remove)
    refs/remotes/origin/ds84182-patch-1                                     stale (use 'git remote prune' to remove)
    refs/remotes/origin/fill_paginated_data_table_card                      stale (use 'git remote prune' to remove)
    refs/remotes/origin/fix                                                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/fix-tap-secondary-after-rejection                   stale (use 'git remote prune' to remove)
    refs/remotes/origin/fix-widget-controller-doc                           stale (use 'git remote prune' to remove)
    refs/remotes/origin/flutter-0.0-candidate.1                             stale (use 'git remote prune' to remove)
    refs/remotes/origin/gallery                                             stale (use 'git remote prune' to remove)
    refs/remotes/origin/generated_explicit_type                             stale (use 'git remote prune' to remove)
    refs/remotes/origin/hot_restart_lowercase_r                             stale (use 'git remote prune' to remove)
    refs/remotes/origin/kevmoo.web_template                                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/keyonghan-deviceLabReadme                           stale (use 'git remote prune' to remove)
    refs/remotes/origin/navigator_change                                    stale (use 'git remote prune' to remove)
    refs/remotes/origin/paginated_data_table_card_width                     stale (use 'git remote prune' to remove)
    refs/remotes/origin/potato620-patch-1                                   stale (use 'git remote prune' to remove)
    refs/remotes/origin/pub-version-mismatch                                stale (use 'git remote prune' to remove)
    refs/remotes/origin/refactor                                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/register_service_public                             stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-28919-composite_elevations                   stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-29010-re_enable_dart2js                      stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-29323-roll_branch                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-30339-tap-buttons                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-30873-revert-30414-remove-hover-pressure     stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-30884-fix-defaultTabController               stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-30919-initial_auth_codes                     stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-30951-roll_branch                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-30991-caretheight                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-30995-revert_engine                          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-31736-roll_build                             stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-31801-revert-30339-tap-buttons               stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-31889-platform_build_api                     stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-32770-modal-dismissed-by-any-button          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-33163-tests_cleanup_1                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-33230-bug_31691                              stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-33297-add_to_app_measurement                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-33688-goldIntegration                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-33794-inline2                                stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-34006-inline2                                stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-34050-fix_file_copy                          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-34823-image-loading-perf-test                stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-36030-tooltip-theme                          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-36465-use_web_config                         stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-37718-zbounds                                stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-37793-tweenequals                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-37845-patch-1                                stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-37962-show-search-app-bar-theme              stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-38828-roll_branch                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-38861-use_frame_timings                      stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-38922-text-docs                              stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-39986-proguard                               stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-40466-double_back                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-40607-animationOverview                      stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-40634-goldenTriageBenchmark                  stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-40710-flutterGold                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-40726-roll_branch                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-40810-jetfier_plugins                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-40995-revert-40925-androidx_default          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-41230-bundle_assemble                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-41338-master                                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-41415-exposeimage                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-41612-aot_linux                              stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-41646-devicelab_vm_service                   stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-41780-fix_build                              stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-42030-revert-41612-aot_linux                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-42236-add_rally_to_examples                  stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-43576-web_compiler                           stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-44315-roll_branch                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-44622-affinity                               stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-45126-gallery_enable_platform_views          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-45354-autocorrection                         stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-45411-screenshot_integration_tests           stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-45951-driver                                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-46010-custom-large-images                    stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-46090-gallery_density                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-46183-issues/43780-take-2                    stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-46640-refactor-dockerfile                    stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-46667-roll_branch                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-46928-rename-proj                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-47027-fix-sliver-layout-assert               stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-47177-setEditingState                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-47254-simplifyChip                           stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-47266-rename-app                             stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-48532-integ_test_embedv2                     stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-49366-revert-48900-theatre2                  stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-49405-frontend_server_4                      stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-49433-precedePadding                         stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-49460-web-benchmarks                         stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-49591-fix_pm_host                            stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-49760-editable                               stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-49827-revert_stuart                          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-49848-alert-dialog-scrollable                stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-50200-full_assemble                          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-50488-cherry-pick-android-check              stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-50502-revert-49480-ink-splash-ripple-mixin   stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-50516-revert-47616-bugfix/snackbar           stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-50775-dialog_padding                         stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-51418-perf1                                  stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-51440-catch-exception                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-51441-revert-51249-live_image_cache          stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-51489-revert-51418-perf1                     stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-51517-thbfix                                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-51886-revert-49848-alert-dialog-scrollable   stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-52055-formatter                              stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-52505-android_device_cleanup                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-52542-android_device_cleanup                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-54497-deflake-run_test2                      stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-55486-max_memory_test                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert-56924-hide_tsi                               stale (use 'git remote prune' to remove)
    refs/remotes/origin/revert_auth_codes                                   stale (use 'git remote prune' to remove)
    refs/remotes/origin/roll_package_test_                                  stale (use 'git remote prune' to remove)
    refs/remotes/origin/sjindel.aar                                         stale (use 'git remote prune' to remove)
    refs/remotes/origin/sjindel.elf                                         stale (use 'git remote prune' to remove)
    refs/remotes/origin/sjindel.type                                        stale (use 'git remote prune' to remove)
    refs/remotes/origin/slider-text-field                                   stale (use 'git remote prune' to remove)
    refs/remotes/origin/time-text-scale                                     stale (use 'git remote prune' to remove)
    refs/remotes/origin/timsneath-api-readme                                stale (use 'git remote prune' to remove)
    refs/remotes/origin/timsneath-precache2                                 stale (use 'git remote prune' to remove)
    refs/remotes/origin/v0.0.1                                              stale (use 'git remote prune' to remove)
    refs/remotes/origin/verbose-put                                         stale (use 'git remote prune' to remove)
    revert-51465-fab-complete-new                                           tracked
    roll_branch                                                             tracked
    stable                                                                  tracked
    v1.12.13-hotfixes                                                       tracked
    v1.4.5-hotfixes                                                         tracked
    v1.4.6-hotfixes                                                         tracked
    v1.4.9-hotfixes                                                         tracked
    v1.5.4-hotfixes                                                         tracked
    v1.7.8-hotfixes                                                         tracked
    v1.9.1-hotfixes                                                         tracked
  Local branches configured for 'git pull':
    beta   merges with remote beta
    master merges with remote master
    stable merges with remote stable
  Local refs configured for 'git push':
    beta   pushes to beta   (up to date)
    master pushes to master (local out of dat
```

**Comment by jmagman** [comment]
> (base) AnsenMBP-4:flutter apple$ 

Did you run that command in your Flutter SDK directory?

**Comment by iyuhang** [comment]
@jmagman 
Yes, in that directory
![image](https://user-images.githubusercontent.com/14176635/81764457-3633e580-9504-11ea-97f9-6c9d687ac6cf.png)

**Comment by iyuhang** [comment]
![image](https://user-images.githubusercontent.com/14176635/81764522-59f72b80-9504-11ea-9aa3-94eafa3fb8a7.png)

**Comment by christopherfujino** [comment]
@iyuhang thanks for filing this. I can repro. The issue is that `git checkout 1.17.0` leaves your git state in a detached head state (in other words, you are no longer on a branch). This means that `flutter upgrade` or `git pull` won't work, because you are not on a branch. However, the error message you saw from the tool is wrong, so we should fix that.

To fix your state, get back on the stable branch:

```
git checkout stable
```

This should check out the latest stable. If you want to revert to the previous stable, you will have to do the following AFTER checking out stable (unfortunately `flutter downgrade` won't work on stable until the **next** release):

```
git reset --hard f139b11
```

**Comment by iyuhang** [comment]
Hi, @christopherfujino
I followed your steps to fix: 

1. git checkout stable

tool shows: Your branch is up to date with 'origin/stable'.

2. git reset --hard f139b11

tool shows:Checking out files: 100% (3763/3763), done.
HEAD is now at f139b1100 update to engine hash that includes Dart 2.7.2 (#53591)

3. flutter doctor:
```
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.14.6 18G87, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.44.2)
[✓] Connected device (1 available)
```

4. flutter upgrade: (It works)

```
Upgrading Flutter from /Users/apple/Flutter/flutter...
....
Flutter 1.17.1 • channel stable • https://github.com/flutter/flutter
Framework • revision f7a6a7906b (23 hours ago) • 2020-05-12 18:39:00 -0700
Engine • revision 6bc433c6b6
Tools • Dart 2.8.2

Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.17.1, on Mac OS X 10.14.6 18G87, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)

[!] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    ! CocoaPods 1.6.1 out of date (1.8.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade:
        sudo gem install cocoapods
[✓] Android Studio (version 3.6)
[✓] VS Code (version 1.44.2)
[✓] Connected device (1 available)
```

5. run flutter upgrade again: (still unable)

`Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter`

6. try again:
```
(base) AnsenMBP-4:flutter apple$ git checkout stable
Already on 'stable'
Your branch is up to date with 'origin/stable'.
(base) AnsenMBP-4:flutter apple$ flutter upgrade
Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter
```

**Comment by christopherfujino** [comment]
@iyuhang What is your goal? Why did you step 5?

**Comment by iyuhang** [comment]
@christopherfujino  Just try to check if flutter upgrade can works 😅

**Comment by iyuhang** [comment]
> unfortunately flutter downgrade won't work on stable until the next release

Do you mean 'flutter upgrade' won't work util the next release by that sentence
If that, I have no question now

**Comment by christopherfujino** [comment]
> > unfortunately flutter downgrade won't work on stable until the next release
> 
> Do you mean 'flutter upgrade' won't work util the next release by that sentence
> If that, I have no question now

No, I mean `flutter downgrade` won't work to go from 1.17 -> 1.12 because the feature was released AFTER 1.12. In order for `flutter downgrade` to work, the tool has to write state to a special file BEFORE doing an upgrade.

**Comment by iyuhang** [comment]
So, how should I upgrade if next version release
Go back to v1.12.13+hotfix.9 then run flutter upgrade there?

**Comment by christopherfujino** [comment]
you should always be able to `flutter upgrade`. where things broke down here was doing `flutter checkout 1.17.0`. if you check out a tag, you will no longer be on a branch, and `flutter upgrade` will be broken.

**Comment by iyuhang** [comment]
while it cannot works now, and I did git checkout stable as the steps above 😢

**Comment by VladyslavBondarenko** [comment]
@iyuhang 
after `git checkout stable` or `flutter channel stable` can you see with `git branch` that your branch is stable?
if it is then `flutter upgrade` should work

**Comment by iyuhang** [comment]
![image](https://user-images.githubusercontent.com/14176635/81919369-5a79ea00-960a-11ea-8522-4abb94e29479.png)

(base) AnsenMBP-4:flutter apple$ git checkout stable
Already on 'stable'
Your branch is up to date with 'origin/stable'.
(base) AnsenMBP-4:flutter apple$ git branch
  beta
  master
* stable
(base) AnsenMBP-4:flutter apple$ flutter upgrade
Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter

**Comment by christopherfujino** [comment]
@iyuhang my PR to surface a better error message is blocked by the tree being red. In lieu, can you try running `git rev-parse --verify @{u}` in the repo and share what the actual error message is?

**Comment by iyuhang** [comment]
@christopherfujino 
The message as follow:
$ git rev-parse --verify @{u}
f7a6a7906be96d2288f5d63a5a54c515a6e987fe

**Comment by christopherfujino** [comment]
> @christopherfujino
> The message as follow:
> $ git rev-parse --verify @{u}
> [f7a6a79](https://github.com/flutter/flutter/commit/f7a6a7906be96d2288f5d63a5a54c515a6e987fe)

Hmm, ok, I'm not sure why you received the error. In any case, my fix to give more specific error handling landed in master, so if you want, you can switch to master and try it out.

**Comment by larssn** [comment]
Why was this closed? Doesn't seem fixed.

Switching to master gives a different error:
```
ProcessException: Process exited abnormally:
From https://github.com/flutter/flutter
 ! [rejected]            v1.7.8+hotfix.1 -> v1.7.8+hotfix.1  (would clobber existing tag)
  Command: git fetch --tags
```

**Comment by christopherfujino** [comment]
> Why was this closed? Doesn't seem fixed.
> 
> Switching to master gives a different error:
> 
> ```
> ProcessException: Process exited abnormally:
> From https://github.com/flutter/flutter
>  ! [rejected]            v1.7.8+hotfix.1 -> v1.7.8+hotfix.1  (would clobber existing tag)
>   Command: git fetch --tags
> ```
> 
> Good ol' `rm -rf flutter` and then recloning the repo fixed whatever broken state my installation had found itself in. 👍🏻

Well that is a different error, which also should be fixed :) I would ask you what `git rev-parse v1.7.8+hotfix.1` returns, but as you wiped out the repo I assume it now points to the upstream revision, `e40273f79707e35e05fdf806be6f6fba7bebd3f8`?

I just created https://github.com/flutter/flutter/issues/57396 to track. I'm keeping that issue separate, as it's possible my "fix" will uncover other git errors which should also be tracked separately.

**Comment by larssn** [comment]
@christopherfujino It does indeed point to e40273f79707e35e05fdf806be6f6fba7bebd3f8 now. I'll keep your issues in mind, and post there if I run into anything again. 👍🏻

**Comment by tvolkert** [comment]
eng approval for cp of #57162 into 1.17

**Comment by pcsosinski** [comment]
removed cp 1.18 label as 1.19 will go to beta soon.

**Comment by Achudh** [comment]
> Its real detailed, much thanks to @guopeng1994
> while I have given up to downgrade now, and try to fix my project issues in new flutter version
> Now the main problem is I cannot run flutter upgrade in 1.17.0
> 
> run flutter upgrade
> 
> shows:
> 
> `Unable to upgrade Flutter: no origin repository configured. Run 'git remote add origin https://github.com/flutter/flutter' in /Users/apple/Flutter/flutter`
> 
> tried:
> 
> 1. flutter channel stable
> 
> 2.git clean -xfd
> git stash save --keep-index
> git stash drop
> git pull
> flutter doctor
> 
> 1. git remote set-url origin https://github.com/flutter/flutter
> 2. git branch --set-upstream-to=origin/stable stable
> 
> none of the above works
> 
> anything I can do to fix it?
> 
> @VladyslavBondarenko @guopeng1994

this worked for me
